cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -407,15 +408,19 @@ mkTracers traceConf tracer = do
             logValue1 = LogValue "txsInMempool" $ PureI $ fromIntegral (msNumTxs tot)
 
             logValue2 :: LOContent a
-            logValue2 = LogValue "txsProcessed" $ PureI $ fromIntegral n
-
-            logValue3 :: LOContent a
-            logValue3 = LogValue "mempoolBytes" $ PureI $ fromIntegral (msNumBytes tot)
+            logValue2 = LogValue "mempoolBytes" $ PureI $ fromIntegral (msNumBytes tot)
 
         meta <- mkLOMeta Critical Confidential
         traceNamedObject tr' (meta, logValue1)
         traceNamedObject tr' (meta, logValue2)
-        traceNamedObject tr' (meta, logValue3)
+ 
+        case mempoolEvent of
+          TraceMempoolRemoveTxs txs _ ->
+            -- TraceMempoolRemoveTxs are previously valid transactions that are no longer valid because of
+            -- changes in the ledger state. These transactions are already removed from the mempool,
+            -- so we can treat them as completely processed.
+            modifyMVar_ txsProcessedCounter $ \counter -> return $ counter + fromIntegral (length txs)
```

Comment:
You're storing a thunk in the `MVar` that holds onto `txs`. As the `+` is lazy, you're accumulating them. This means that you're holding on to *all* the transactions that are removed.

The only thing forcing the thunk is `when (txsProcessed /= txsNum)`, so you're only holding onto the transactions that were removed in a single second. So not a huge space leak, but many small ones can still make the ship sink.

So: force that `counter` before storing it! Alternatively, would `atomicModifyIORef'` suffice here? And why an `Integer`? What about `Word64` or `Word`?

Reply:
> And why an `Integer`?

Because we want to trace that value (from `MVar`) as `LogValue` with `PureI` (which takes `Integer`).

Anyway, @mrBliss, this is "not-finished" solution. I mean, we need some additional information 
to trace (for example, the number of forged blocks), so we'll store a structure in this `MVar`, not a single value. Moreover, it will probably be `TVar` instead of `MVar`.

Reply:
> > And why an `Integer`?
> 
> Because we want to trace that value (from `MVar`) as `LogValue` with `PureI` (which takes `Integer`).

Alright.

> 
> Anyway, @mrBliss, this is "not-finished" solution. I mean, we need some additional information to trace (for example, the number of forged blocks, please see CAD-803). So we'll store a structure in this `MVar`, not a single value. Moreover, it will probably be `TVar` instead of `MVar`. This PR is solving one single bug.

I don't understand why you then merge this "not-finished" solution. You just added a small space leak (reset every second, fortunately) :man_shrugging:

Reply:
@denisshevchenko We must fix this as a high priority task.

Reply:
> I don't understand why you then merge this "not-finished" solution.

Because it solves a bug. An old bug, both in TUI and in RTView service.